flake.nix

```diff
@@ -0,0 +1,45 @@
+{
+  description = "A Nix-flake-based development environment for Cursorless";
+
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+
+  outputs = { self, nixpkgs }:
+    let
+      overlays = [
+        (final: prev: rec {
+          nodejs = prev.nodejs-18_x;
+          pnpm = prev.nodePackages.pnpm;
+          yarn = (prev.yarn.override { inherit nodejs; });
+        })
+      ];
+      supportedSystems =
+        [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ];
+      forEachSupportedSystem = f:
+        nixpkgs.lib.genAttrs supportedSystems (system:
+          f {
+            pkgs = import nixpkgs {
+              inherit overlays system;
+              config.allowUnfree = true;
+            };
+          });
+    in {
+      devShells = forEachSupportedSystem ({ pkgs }: {
+        default = pkgs.mkShell {
+          packages = with pkgs; [
+            nodejs
+            pnpm
+            python310
```

Comment:
We should probably set this to match the Python version in our pyproject.toml. Not ideal that we now have yet another source of truth for a Python version, but I'm not sure there's a way around that. Does nix  have support for reading from PyProject.toml?

Reply:
That's a good idea. I just uses python310 here because I copied what worked on vscode-parse-tree iirc 😅 I know for actual package building on nix you get get it to read that config file, so I suspect there is something similar for prepping the dev shell. I'll play around with it later.

Reply:
fixed. i pull it out of the ruff part of pyproject.toml since there isn't an explicit version elsewhere. 

also, wow somehow 4 months passed... i've been using the flake for a devshell on all my PRs (4?) for the last 4 months, and has worked well fwiw. I still haven't bothered actually adding the build part in, so that will come in a follow up PR some day (hopefully not another 4 months)